src/utils/dezswap.ts

```diff
@@ -180,16 +221,66 @@ export const generateWithdrawLiquidityMsg = (
   );
 
 export const generateSwapMsg = (
+  isSimulate: boolean,
   senderAddress: string,
   contractAddress: string,
   fromAssetAddress: string,
   amount: string,
+  authSequence: bigint,
   beliefPrice: Numeric.Input,
   maxSpread = "0.1",
   txDeadlineSeconds = 1200,
 ) => {
   const maxSpreadFixed = `${(parseFloat(maxSpread) / 100).toFixed(4)}`;
-  if (AccAddress.validate(fromAssetAddress)) {
+  const isCW20 = AccAddress.validate(fromAssetAddress);
+  if (!isSimulate) {
+    if (isCW20) {
+      const sendMsg = window.btoa(
+        JSON.stringify({
+          swap: {
+            max_spread: `${maxSpreadFixed}`,
+            belief_price: `${beliefPrice}`,
+            deadline: Number(
+              Number((Date.now() / 1000).toFixed(0)) + txDeadlineSeconds,
+            ),
+          },
+        }),
+      );
+
+      return new BeforeMsgExecuteContract(
```

Comment:
Why use xpla.js type? 
If you used `MsgExecuteContract.fromPartial(` here, you don't need the conditional statement below.